### PR TITLE
fix(useEVMTxn): disable gas estimate if tx disabled

### PR DESCRIPTION
**Pull request by vbstreetz**

### CODE REVIEW

File: packages/queries/src/mutations/useContractTxn.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ const useContractTxn = (
 	txnOptions: Partial<ethers.providers.TransactionRequest> = {},
 	options?: UseEVMTxnOptions
 ) => {
-	if (contract != null && method != null) {
+	if (contract != null && method != null && options?.enabled) {
```
Comment (dbeal-eth) [inline packages/queries/src/mutations/useContractTxn.ts:14]: why is this needed? also what happensi f `options` is not defined (as it often isnt)?

File: packages/queries/src/mutations/useEVMTxn.ts [hunk 2]
```diff
@@ -60,6 +60,10 @@ const useEVMTxn = (
 
 		setErrorMessage(null);
 
+		if (!options.enabled) {
```
Comment (dbeal-eth) [inline packages/queries/src/mutations/useEVMTxn.ts:63]: would probably be better to just do `if(options.enabled) estimateGas().then()...`

**Review by dbeal-eth — CHANGES_REQUESTED** [review]

File: packages/queries/src/mutations/useContractTxn.ts (hunk 1 above)
Comment (vbstreetz) [reply packages/queries/src/mutations/useContractTxn.ts:14]: this breaks if args are not valid https://github.com/Synthetixio/js-monorepo/blob/master/packages/queries/src/mutations/useContractTxn.ts#L19

File: packages/queries/src/mutations/useEVMTxn.ts (hunk 2 above)
Comment (vbstreetz) [reply packages/queries/src/mutations/useEVMTxn.ts:63]: i like early returns but fixed :)

File: packages/queries/src/mutations/useContractTxn.ts (hunk 1 above)
Comment (dbeal-eth) [reply packages/queries/src/mutations/useContractTxn.ts]: really? interesting. good catch :+1:

**Review by dbeal-eth — APPROVED** [review]
there are a couple "while you are at it" changes I would like to request but one thing at a time is probably best.

thanks for your work on this :+1:

**Comment by vbstreetz** [comment]
> there are a couple "while you are at it" changes I would like to request but one thing at a time is probably best.
> 
> thanks for your work on this 👍

say more?

File: packages/queries/src/mutations/useContractTxn.ts (hunk 1 above)
Comment (vbstreetz) [reply packages/queries/src/mutations/useContractTxn.ts]: btw, i updated to https://github.com/Synthetixio/js-monorepo/blob/fix/useEvmTxn/packages/queries/src/mutations/useContractTxn.ts#L14

**Comment by dbeal-eth** [comment]
if you have time/desire, some fixes that would be helpful are:
* fixing `UseEVMOptions` such that all the arguments are optional instead of required all at once
* improving errored transaction handling (right now its not always good at parsing the errors to set for `txn.errorMessage`)
* adding EIP-1559 support (which would add EIP-1559 to all of our downstream apps, which would be nice)

File: packages/queries/src/mutations/useContractTxn.ts (hunk 1 above)
Comment (vbstreetz) [reply packages/queries/src/mutations/useContractTxn.ts]: > if you have time/desire, some fixes that would be helpful are:
> 
> * fixing `UseEVMOptions` such that all the arguments are optional instead of required all at once
> * improving errored transaction handling (right now its not always good at parsing the errors to set for `txn.errorMessage`)
> * adding EIP-1559 support (which would add EIP-1559 to all of our downstream apps, which would be nice)

awesome. tackling these in a new or..
